### PR TITLE
Fixes #679.

### LINKED ISSUE #679: Add checkstyle check on pull request validation
Before pull request merge, we need to add checkstyle to check the code format and style.
Define a checkstyle template and add it to maven validation phase is a good practice.

We can use ShardingSphere's checkstyle template directly, there is in:
https://github.com/apache/incubator-shardingsphere/blob/dev/src/resources/checkstyle_ci.xml

### PR DESCRIPTION
**Pull request by Leonidasz**
Fixes #679.

Changes proposed in this pull request:
- Add new checkstyle to project.
- Update code with new checkstyle.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -521,7 +540,8 @@
                 <artifactId>maven-checkstyle-plugin</artifactId>
                 <version>${maven-checkstyle-plugin.version}</version>
                 <configuration>
-                    <configLocation>src/main/resources/dd_checks.xml</configLocation>
+                    <configLocation>src/main/resources/checkstyle_ci.xml</configLocation>
+                    <includeTestSourceDirectory>false</includeTestSourceDirectory>
```
Comment (terrymanu) [inline pom.xml]: Should include includeTestSourceDirectory too

File: pom.xml [hunk 2]
```diff
@@ -431,7 +432,25 @@
...
                 </executions>
             </plugin>
-             <plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${maven-checkstyle-plugin.version}</version>
+                <configuration>
+                    <configLocation>src/main/resources/checkstyle_ci.xml</configLocation>
+                    <includeTestSourceDirectory>false</includeTestSourceDirectory>
```
Comment (terrymanu) [inline pom.xml]: Should include includeTestSourceDirectory too

File: pom.xml [hunk 3]
```diff
@@ -521,7 +540,8 @@
                 <artifactId>maven-checkstyle-plugin</artifactId>
                 <version>${maven-checkstyle-plugin.version}</version>
                 <configuration>
-                    <configLocation>src/main/resources/dd_checks.xml</configLocation>
+                    <configLocation>src/main/resources/checkstyle_ci.xml</configLocation>
```
Comment (terrymanu) [inline pom.xml]: For report should use `checkstyle.xml`, not `checkstyle_ci.xml`.
This pr add 2 template files of checkstyle, is it?

File: pom.xml [hunk 4]
```diff
@@ -263,7 +264,7 @@
             </dependency>
         </dependencies>
     </dependencyManagement>
-    
+
```
Comment (terrymanu) [inline pom.xml]: Should not change original code indent.
Please disable code auto formatting on code commit.

**Review by terrymanu — CHANGES_REQUESTED** [review]
Should not change original code indent.
Please disable code auto formatting on code commit.

File: pom.xml (hunk 1 above)
Comment (Leonidasz) [reply pom.xml]: Ok, got it.

File: pom.xml (hunk 3 above)
Comment (Leonidasz) [reply pom.xml]: Yes, we add 2 temlate files of checkstyle. I thought checkstyle.xml was for the IDE. I will change the setting.

File: pom.xml (hunk 2 above)
Comment (Leonidasz) [reply pom.xml]: Ok, got it.

File: pom.xml (hunk 4 above)
Comment (Leonidasz) [reply pom.xml]: Ok, got it.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**